Source/Plugins/Statistics/StatisticsFilters/FindSizes.cpp

```diff
@@ -204,13 +204,17 @@ void FindSizes::findSizesImage(ImageGeom::Pointer image)
   size_t totalPoints = m_FeatureIdsPtr.lock()->getNumberOfTuples();
   size_t numfeatures = m_VolumesPtr.lock()->getNumberOfTuples();
 
-  DataArray<float>::Pointer m_FeatureCounts = DataArray<float>::CreateArray(numfeatures, "_INTERNAL_USE_ONLY_FeatureCounts");
+  // Previous versions of this filter used floats for their calculations.  However, with large data sets, the feature count 
+  // surpasses the maximum integer value that a floating point number can contain.  Converting over to doubles temporarily 
+  // aleviates the problem, but a more in-depth algorithm to prevent or minimize inaccuracies when dealing with large data sets
+  // should be implemented in the future.
+  DataArray<double>::Pointer m_FeatureCounts = DataArray<double>::CreateArray(numfeatures, "_INTERNAL_USE_ONLY_FeatureCounts");
   m_FeatureCounts->initializeWithZeros();
-  float* featurecounts = m_FeatureCounts->getPointer(0);
+  double* featurecounts = m_FeatureCounts->getPointer(0);
 
-  float rad = 0.0f;
-  float diameter = 0.0f;
-  float res_scalar = 0.0f;
+  double rad = 0.0f;
+  double diameter = 0.0f;
+  double res_scalar = 0.0f;
 
   for(size_t j = 0; j < totalPoints; j++)
   {
```

Comment:
Can you throw this bit into the code to sanity check the largest number that we can deal with:

  // We know we may have issues if we ever have a feature with more than 
  //9007199254740992 voxels (9 Petabytes worth) so check that real quick  
  for(size_t i = 1; i < numfeatures; i++)
  {
    if(featurecounts[i] > static_cast<double>(9007199254740992))
    {
      setErrorCondition(-10201);
      QString ss = QObject::tr("Feature Id %1 has more than 9,007,199,254,740,992 voxels which will cause math errors when calculating the size value.").arg(i);
      notifyErrorMessage(getHumanLabel(), ss, getErrorCondition());
      return;
    }
  }
 
Just after line 223 which should be   std::tie(xRes, yRes, zRes) = image->getResolution();